File: MP1/SIS.py
from dataclasses import dataclass


@dataclass
class Student:
    name: str
    age: int
    year_level: int
    course: str
    gwa: float
    grade_level: str


def determine_grade_level(year_level: int) -> str:
    match year_level:
        case 1:
            return "Freshman"
        case 2:
            return "Sophomore"
        case 3:
            return "Junior"
        case 4:
            return "Senior"
        case _:
            return "Unknown"


def display_student(student: Student) -> None:
    print("\n--- Student Information ---")
    print(f"Name: {student.name}")
    print(f"Age: {student.age}")
    print(f"Year Level: {student.year_level}")
    print(f"Grade Level: {student.grade_level}")
    print(f"Course: {student.course}")
    print(f"GWA: {student.gwa:.6f}")


def main() -> None:
    name = input("Enter Name: ")
    age = int(input("Enter Age: "))
    year_level = int(input("Enter Year Level (1-4): "))
    course = input("Enter Course: ")
    gwa = float(input("Enter GWA: "))

    grade_level = determine_grade_level(year_level)

    student = Student(
        name=name,
        age=age,
        year_level=year_level,
        course=course,
        gwa=gwa,
        grade_level=grade_level,
    )

    display_student(student)


if __name__ == "__main__":
    main()

File: MP1/SIS.cpp
#include <iostream>
#include <string>

struct Student {
    std::string name;
    int age;
    int yearLevel;
    std::string course;
    float gwa;
    std::string gradeLevel;
};

std::string DetermineGradeLevel(int yearLevel);
void DisplayStudent(const Student& student);

int main() {
    Student student;

    std::cout << "Enter Name: ";
    std::getline(std::cin, student.name);

    std::cout << "Enter Age: ";
    std::cin >> student.age;

    std::cout << "Enter Year Level (1-4): ";
    std::cin >> student.yearLevel;

    std::cin.ignore();
    std::cout << "Enter Course: ";
    std::getline(std::cin, student.course);

    std::cout << "Enter GWA: ";
    std::cin >> student.gwa;

    student.gradeLevel = DetermineGradeLevel(student.yearLevel);

    DisplayStudent(student);

    return 0;
}

std::string DetermineGradeLevel(int yearLevel) {
    switch (yearLevel) {
        case 1:
            return "Freshman";
        case 2:
            return "Sophomore";
        case 3:
            return "Junior";
        case 4:
            return "Senior";
        default:
            return "Unknown";
    }
}

void DisplayStudent(const Student& student) {
    std::cout << "\n--- Student Information ---" << std::endl;
    std::cout << "Name: " << student.name << std::endl;
    std::cout << "Age: " << student.age << std::endl;
    std::cout << "Year Level: " << student.yearLevel << std::endl;
    std::cout << "Grade Level: " << student.gradeLevel << std::endl;
    std::cout << "Course: " << student.course << std::endl;
    std::cout << "GWA: " << student.gwa << std::endl;
}